### PR TITLE
Main PR for 2022-2023

### PR DESCRIPTION
**Pull request by raghavauppuluri13**
integration pr for 2022-2023

### CODE REVIEW

File: lunabot_embedded/src/teensy_driver_node.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+void effort_cb(const lunabot_msgs::RobotEffort &msg) {
+    effort.lead_screw = msg.lead_screw;
+    effort.lin_act = msg.lin_act;
+    effort.left_drive = msg.left_drive;
+    effort.right_drive = msg.right_drive;
+    effort.excavate = msg.excavate;
+    effort.deposit = msg.deposit;
+}
+
+void publish(const ros::TimerEvent&) {
```
Comment (raghavauppuluri13) [inline lunabot_embedded/src/teensy_driver_node.cpp:58]: Make more clear on what publish does (update function name?)

File: lunabot_control/scripts/manual_controller.py [hunk 2]
```diff
@@ -3,44 +3,91 @@
...
+1 u/d stick left
+2 LT
+3 l/r stick right
+"""
 from lunabot_msgs.msg import RobotEffort
 
 
 def constrain(joy_in):
-    return np.int8(min(joy_in * 128, 127))
+    return np.int8(joy_in * 127)
```
Comment (ashviniyer21) [inline lunabot_control/scripts/manual_controller.py:30]: Constrain between -1 and 1 before multiplying by 127 (just for safety)

File: lunabot_msgs/msg/RobotState.msg
Comment (ashviniyer21) [inline lunabot_msgs/msg/RobotState.msg:1]: Rename to be more clear (robot sensors?) State makes it seem like it should store position after SLAM

**Review by ashviniyer21 — APPROVED** [review]
Added couple of minor comments, overall looks good to merge

File: lunabot_embedded/firmware/update_board_manager.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import os
+import subprocess
+import tarfile
+import shutil
+import yaml
+
+# Step 4: Make the .arduino directory if it doesn't exist
+yaml_file_path = os.path.join(os.path.expanduser('~'), '.arduino15/arduino-cli.yaml')
```
Comment (ashviniyer21) [inline lunabot_embedded/firmware/update_board_manager.py:8]: if __name__ == "__main__" this code? Didn't look at all files but same feedback if this exists in other places

**Comment by raghavauppuluri13** [comment]
> LGTM. Left 1 minor comment (not even sure if that's your code or if it's library code lol). Didn't look in too much detail but trusting feedback from `feature/autonomy` PR was addressed properly.

fixed! will merge then.

**Comment by raghavauppuluri13** [comment]
> LGTM. Left 1 minor comment (not even sure if that's your code or if it's library code lol). Didn't look in too much detail but trusting feedback from `feature/autonomy` PR was addressed properly.

for clarity, these were the ones not addressed in this PR (non-blocking):

higher priority fixes

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1451099217

nits + docs updates

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1421110396

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1264551454

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1264551573

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1451519852

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1264553342

File: readme.md
Comment (Isaac-F24) [inline readme.md:1]: https://wiki.purduearc.com/wiki/tutorials/setup-ros is a dead link.

Comment (raghavauppuluri13) [reply readme.md:1]: good catch! should be fixed now

**Review by ashviniyer21 — CHANGES_REQUESTED** [review]
Blocking PR merge until changes are tested on the robot

**Comment by raghavauppuluri13** [comment]
blocked by https://github.com/PurdueLunabotics/purdue_lunabotics/pull/55. once that's tested in sim and real, can merge this.

File: readme.md [hunk 4]
```diff
@@ -16,10 +34,12 @@ This tutorial assumes you have your `catkin_ws` initialized in your home directo
 cd ~/catkin_ws/src
 git clone https://github.com/PurdueLunabotics/purdue_lunabotics.git
```
Comment (Isaac-F24) [inline readme.md:35]: Suggestion: add recurse submodules instructions

File: lunabot_embedded/firmware/update_board_manager.py (hunk 3 above)
Comment (raghavauppuluri13) [reply lunabot_embedded/firmware/update_board_manager.py]: done

File: lunabot_msgs/msg/RobotState.msg
Comment (raghavauppuluri13) [reply lunabot_msgs/msg/RobotState.msg:1]: done

File: readme.md (hunk 4 above)
Comment (raghavauppuluri13) [reply readme.md]: done

**Pull request merged**